src/components/ReanimatedSwipeable.tsx

```diff
@@ -523,16 +557,30 @@ const Swipeable = forwardRef<SwipeableMethods, SwipeableProps>(
             ? 'left'
             : 'right';
 
-        if (rowState.value === 0 && props.onSwipeableOpenStartDrag) {
-          runOnJS(props.onSwipeableOpenStartDrag)(direction);
-        } else if (props.onSwipeableCloseStartDrag) {
-          runOnJS(props.onSwipeableCloseStartDrag)(direction);
+        if (rowState.value === 0 && onSwipeableOpenStartDrag) {
+          runOnJS(onSwipeableOpenStartDrag)(direction);
+        } else if (rowState.value !== 0 && onSwipeableCloseStartDrag) {
+          runOnJS(onSwipeableCloseStartDrag)(direction);
         }
         updateAnimatedEvent();
       })
-      .onEnd((event) => {
-        handleRelease(event);
-      });
+      .onEnd(
+        (event: GestureStateChangeEvent<PanGestureHandlerEventPayload>) => {
+          handleRelease(event);
+        }
+      );
+
+    const remainingProps = { ...props };
+
+    panGesture.config = { ...remainingProps } as Extract<
+      SwipeableProps,
+      typeof panGesture.config
+    >;
+
+    tapGesture.config = { ...remainingProps } as Extract<
+      SwipeableProps,
+      typeof tapGesture.config
+    >;
```

Comment:
```suggestion
```

That's a bad idea - you're effectively removing the config set above to assign props to it. The only prop that it would make sense for is `enableTrackpadTwoFingerGesture` and you can just set it on the gesture directly:
```jsx
panGesture.enableTrackpadTwoFingerGesture(props.enableTrackpadTwoFingerGesture)
```

Also you can remove `useNativeAnimations` from `SwipeableProps` since all Reanimated animations are native.

Reply:
I think it wasn't so bad, it automatically recognized all suitable props, making room for change in the future as well, and applied them before any of them were explicitly set. It's also quite readable.
But I replaced it as per your recommendation.